src/power_grid_model/utils.py

```diff
@@ -14,7 +14,8 @@
 
 import numpy as np
 
-from power_grid_model.core.power_grid_dataset import get_dataset_type
+from power_grid_model._utils import get_and_verify_batch_sizes
+from power_grid_model.core.power_grid_dataset import CConstDataset, get_dataset_type
```

Comment:
the `CConstDataset` is a low-level wrapper of the C API functionality and is not something the user should care about

```suggestion
from power_grid_model.core.power_grid_dataset import get_dataset_type
```

Reply:
What is the official way of constructing a sparse dataset? I imported CConstDataset only for the sake of sparse dataset. If there is a proper class to use, it indeed shouldn't appear.

Reply:
there's no official way but probably, it's useful to create it as follows:

```py
"sym_power_sensor": {
  "indptr": np.ndarray([0, 0, 1]),
  "data": initialize_array("update", "sym_power_sensor", (1)),
}
```
as opposed to dense
```py
"sym_power_sensor": initialize_array("update", "sym_power_sensor", (2, 1))
```